### ISSUE TITLE
OpenVPN Client configuration (0.9.2)

### ISSUE THREAD

**Issue by TheAlbanian** [d3vilh/openvpn-ui#30]
hello i want from template 2 extra setting in .opvn file.
askpass UserAuth.file
log /etc/openvpn/openvpn.log

here no option add extra settings
![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/e14912da-71f7-4a45-8a4c-62574248fba4)

directly from /root/openvpn/config/client.conf dont load the config.

also from panel config no load extra config
![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/42ed882b-cb9f-44ce-8237-731b9fdf52d1)

**Comment by d3vilh** [comment]
Hi @TheAlbanian,
Thanks for reporting this issue. 
In the next release I'll fix template issue and add 3 more custom fields to use with the template.
At the moment, as workaround you can manually take certificates from `./openvpn-server/clients` there are `*.ovpn` files which been created during certificates generation, during this process OpenVPN-UI using `config.conf`.

**Comment by TheAlbanian** [comment]
hello @d3vilh 
i have small issue.. 

i'm try to add some config

auth-user-pass-verify plugins/checkpsw.sh via-env
client-cert-not-required
username-as-common-name
script-security 3 execve        

i get this error..
```
Options error: --auth-user-pass-verify script fails with 'plugins/checkpsw.sh': No such file or directory (errno=2)
Options error: Please correct this error.
Use --help for more information.
```

i try change many folders.
/root/openvpn/plugins/ also in no root folder..

file exist checkpsw.sh need chmod 0755 
chmod +x /root/openvpn/checkpsw.sh

honestly first time

**Comment by d3vilh** [comment]
> i try change many folders.

which openvpn-server docker container do you use?

**Comment by TheAlbanian** [comment]
> > i try change many folders.
> 
> which openvpn-server docker container do you use?

openvpn-aws

![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/061b9134-5e45-474f-a611-b6ffa5d8759a)

**Comment by d3vilh** [comment]
> > > i try change many folders.
> > 
> > 
> > which openvpn-server docker container do you use?
> 
> openvpn-aws
> 

you can put your authentication script in `~/openvpn/` directory of your main OS and add into `~/openvpn/config/server.conf` this line:  `auth-user-pass-verify /opt/app/checkpsw.sh via-env` then restart server it should works fine.

**Comment by TheAlbanian** [comment]
> auth-user-pass-verify /opt/app/checkpsw.sh via-env

![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/49fd9afa-1b94-4156-8ffd-d4db63146bcd)
![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/aaa679ea-7c57-4683-bfc7-58c7bb08a1b9)
![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/0b925c04-530d-43f0-9b8a-383e737dba42)

**Comment by d3vilh** [comment]
OK, I see 2 ways to fix this:
## Easy way
Put file in the `/root/openvpn/config/checkpsw.sh` and update `server.conf` with this line: `auth-user-pass-verify /etc/openvpn/config/checkpsw.sh via-env`

## The most correct way
update docker-compose volumes configuration with your script (last line in config below):
```shell
~/openvpn $ cat docker-compose.yml
---
version: "3.5"

services:
    openvpn:
       container_name: openvpn
       build: ./openvpn-docker
       privileged: true
       ports:
          - "1194:1194/udp"
       environment:
           TRUST_SUB: 10.0.70.0/24
           GUEST_SUB: 10.0.71.0/24
           HOME_SUB: 192.168.0.0/24
       volumes:
           - ./pki:/etc/openvpn/pki
           - ./clients:/etc/openvpn/clients
           - ./config:/etc/openvpn/config
           - ./staticclients:/etc/openvpn/staticclients
           - ./log:/var/log/openvpn
           - ./fw-rules.sh:/opt/app/fw-rules.sh
           - ./checkpsw.sh:/opt/app/checkpsw.sh
```

and keep `auth-user-pass-verify /opt/app/checkpsw.sh via-env` in `server.conf`.

You have to restart container to apply the change:

```shell
cd ~openvpn
docker restart openvpn
```

**Comment by TheAlbanian** [comment]
done... 
i'm sorry bro i dont know in dacker need mount volume points 
![image](https://github.com/d3vilh/openvpn-ui/assets/57067508/8d76c299-3d75-418d-a581-06225023f532)

**Comment by BliNeR-KeY** [comment]
@d3vilh
What do you think to add a. Radius plugin connected in openvpn-ui database? 
https://github.com/rakasatria/ovpn-radius

**Comment by d3vilh** [comment]
> @d3vilh What do you think to add a. Radius plugin connected in openvpn-ui database? https://github.com/rakasatria/ovpn-radius

I'll include 2FA based on oath-tool in release `0.9.3` which will be out later this week.
Radius can be in the roadmap as well.

**Comment by BliNeR-KeY** [comment]
> > @d3vilh What do you think to add a. Radius plugin connected in openvpn-ui database? https://github.com/rakasatria/ovpn-radius
> 
> I'll include 2FA based on oath-tool in release `0.9.3` which will be out later this week. Radius can be in the roadmap as well.

i have build 0.9.3 i'm try to connect by radius no get success .. from .OVPN i see same Classic Auth... 

i think by radius username/ password 
[-----BEGIN ENCRYPTED PRIVATE KEY----- ] no need be included.

example.... like surfshark

example certs files https://my.surfshark.com/vpn/api/v1/server/configurations

only auth-user-pass ... user/pass

**Comment by BliNeR-KeY** [comment]
@d3vilh i  hope give this. is very Nice.

> > > @d3vilh What do you think to add a. Radius plugin connected in openvpn-ui database? https://github.com/rakasatria/ovpn-radius
> > 
> > 
> > I'll include 2FA based on oath-tool in release `0.9.3` which will be out later this week. Radius can be in the roadmap as well.
> 
> i have build 0.9.3 i'm try to connect by radius no get success .. from .OVPN i see same Classic Auth...
> 
> i think by radius username/ password [-----BEGIN ENCRYPTED PRIVATE KEY----- ] no need be included.
> 
> example.... like surfshark
> 
> example certs files https://my.surfshark.com/vpn/api/v1/server/configurations
> 
> only auth-user-pass ... user/pass

**Comment by babyhuey** [comment]
I'm also running into issues where when I make changes on the openvpn client page, it doesn't appear to be getting set anywhere. 
I am trying to comment out the `#redirect-gateway def1` and it shows as commented on the "view" and the config page, but when I create a user, it is still uncommented. 

Also need a way to change the group from `nogroup` to `nobody`

(Do you want me to create a separate issue with these problems?)

**Comment by d3vilh** [comment]
> I'm also running into issues where when I make changes on the openvpn client page, it doesn't appear to be getting set anywhere. I am trying to comment out the `#redirect-gateway def1` and it shows as commented on the "view" and the config page, but when I create a user, it is still uncommented.
> 
> Also need a way to change the group from `nogroup` to `nobody`
> 
> (Do you want me to create a separate issue with these problems?)

Thanks for the reporting, I'll focus on testing and bug fix tomorrow, no new issue is necessary.

I drop the client template editing option via the editor in the latest push(you still can view, but read only), as it is contradicts with the design. 
That is why I'll include all the client config options to be editable with the main template.

**Comment by d3vilh** [comment]
It is fixed now (in [0.9.4](https://github.com/d3vilh/openvpn-ui/releases/tag/0.9.4)) and you have all Client's template options available for editing.
